src/library_fs.js

```diff
@@ -1135,12 +1134,11 @@ mergeInto(LibraryManager.library, {
       }
       if (stream.flags & {{{ cDefine('O_APPEND') }}}) {
         // seek to the end before writing in append mode
-        position = FS.llseek(stream, 0, {{{ cDefine('SEEK_END') }}});
+        FS.llseek(stream, 0, {{{ cDefine('SEEK_END') }}});
       }
-      var seeking = true;
-      if (typeof position === 'undefined') {
+      var seeking = typeof position !== 'undefined';
+      if (!seeking) {
         position = stream.position;
```

Comment:
@kripken I'm talking about this line, it will use `stream.position` set by `FS.llseek` when we are in append mode.

Reply:
(It seems I'm tired too)

Reply:
Still, on line 1139 (2 lines before, where we do `typeof position`, we are looking at `position` and the change on line 1137 (to not set `position`) can influence that? So the 1137 change can change the behavior, or at least it looks that way.

But if I understood your last comment above, we just use position to see if we are in seek mode? So maybe we can refactor the check to put something like `var inSeekMode = typeof position === 'undefined';`? (can leave this for a later PR)

Reply:
>So the 1137 change can change the behavior, or at least it looks that way.

Yes, we changed (fixed) the behavior, as it was previously broken and caused incorrect llseek behavior after append mode write. I added `test_fs_llseek` to make sure it won't be broken again.

I agree that further refactoring can be done in a later PR.